File: vanilla_js/line_number.js
/**
 *  File        :   line_number.js
 *  Explain     :   Calculates the line number of the text given as input.
 *  Author      :   Mehmet Ekemen
 *  Nickname    :   `hun
 *  Email       :   ekemenms@hotmail.com
 *  Date        :   08-03-2021 (MM-DD-YYYY)  (Updated)
 *  Github      :   github.com/hun756
 *  Resource    :   https://github.com/hun756/TextTool
**/


/**
 *  @brief
 *      This function calculates the number of lines 
 *      in the element containing the given text as input.
 * 
 *  @param{inputElement}
 *      An input element is an element that contains text. 
 *      The value in this item will be processed by the program.     
 * 
 *  @param{resultElement} 
 *      Line number produces a value of type "number". 
 *      This generated value is printed in the 
 *      content of the "resultElement".
**/
let lineNumber = (inputElement, resultElement) => {
    inputElement.addEventListener('input', () => {
        let text = inputElement.value;
    
        let lines = text.split(/\r|\r\n|\n/);
        let count = lines.length;
    
        resultElement.textContent = `Total Line Number is : ${count}`;
    });
}

File: cpp/cheerp/line_number_calculator/line_number.cpp
/**
 *  @file line_number.cpp
 *  @author Mehmet Ekemen <ekemenms@gmail.com>, <ekemenms@hotmail.com>
 *  @brief 
 *  @version 0.1
 *  @date 2021-08-03
 *  
 *  @copyright Copyright (c) 2021
 *  
 *  @link https://leaningtech.com/cheerp/
**/

#include <string>
#include <cheerp/clientlib.h>
#include <cheerp/client.h>

/**
 *  @brief 
 *      Application Starting Point
 */
void webMain()
{

    client::HTMLInputElement *inputElement =
        static_cast<client::HTMLInputElement *>(client::document.getElementById("inputText"));

  
    client::Element *resultElement = client::document.getElementById("result");


    inputElement->addEventListener(
        "input",
        cheerp::Callback([inputElement, resultElement]() -> void
                        {
                            client::String *text = inputElement->get_value();

                            client::RegExp *regex = new client::RegExp("\r\n|\r|\n");

                            auto lines = text->split(regex);

                            int count = lines->get_length();

                            std::string _resStr = "Lines of number is : " + std::to_string(count);
                            resultElement->set_textContent(_resStr.c_str());
                        }
        )
    );
}